File: src/main/java/com/indiepost/controller/admin/AdminAnalyticsController.java
package com.indiepost.controller.admin;

import com.indiepost.dto.stat.*;
import com.indiepost.service.AnalyticsService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

/**
 * Created by jake on 17. 4. 29.
 */
@RestController
@RequestMapping(value = "/admin/stat", produces = {"application/json; charset=UTF-8"})
public class AdminAnalyticsController {

    private final AnalyticsService analyticsService;

    @Autowired
    public AdminAnalyticsController(AnalyticsService analyticsService) {
        this.analyticsService = analyticsService;
    }

    @PostMapping
    public OverviewStats getOverviewStats(@RequestBody PeriodDto periodDto) {
        return analyticsService.getOverviewStats(periodDto);
    }

    @PostMapping("/tops")
    public TopStats getTopStats(@RequestBody PeriodDto periodDto) {
        return analyticsService.getTopStats(periodDto);
    }

    @PostMapping("/recent-and-old")
    public RecentAndOldPostStats getRecentAndOldPostStats(@RequestBody PeriodDto periodDto) {
        return analyticsService.getRecentAndOldPostStats(periodDto);
    }

    @GetMapping("/posts")
    public PostStatsDto getAllPostStats() {
        return analyticsService.getCachedPostStats();
    }

    @PostMapping("/posts")
    public PostStatsDto getPostStats(@RequestBody PeriodDto periodDto) {
        return analyticsService.getPostStats(periodDto);
    }
}

File: src/main/kotlin/com/indiepost/controller/admin/AdminAnalyticsController.kt
package com.indiepost.controller.admin

import com.indiepost.dto.analytics.*
import com.indiepost.service.AnalyticsService
import org.springframework.web.bind.annotation.*

import javax.inject.Inject

/**
 * Created by jake on 17. 4. 29.
 */
@RestController
@RequestMapping(value = ["/admin/stat"], produces = ["application/json; charset=UTF-8"])
class AdminAnalyticsController @Inject constructor(private val analyticsService: AnalyticsService) {

    @PostMapping
    fun getOverviewStats(@RequestBody periodDto: PeriodDto): OverviewStats {
        return analyticsService.getOverviewStats(periodDto)
    }

    @PostMapping("/tops")
    fun getTopStats(@RequestBody periodDto: PeriodDto): TopStats {
        return analyticsService.getTopStats(periodDto)
    }

    @PostMapping("/recent-and-old")
    fun getRecentAndOldPostStats(@RequestBody periodDto: PeriodDto): RecentAndOldPostStats {
        return analyticsService.getRecentAndOldPostStats(periodDto)
    }

    @GetMapping("/posts")
    fun getAllPostStats(): PostStatsDto? {
        return analyticsService.getCachedPostStats()
    }

    @PostMapping("/posts")
    fun getPostStats(@RequestBody periodDto: PeriodDto): PostStatsDto {
        return analyticsService.getPostStats(periodDto)
    }
}
